src/MagnumPlugins/TinyGltfImporter/TinyGltfImporter.cpp

```diff
@@ -0,0 +1,484 @@
+}
+
+Int TinyGltfImporter::doDefaultScene() {
+    return _d->model.defaultScene;
+}
+
+Containers::Optional<CameraData> TinyGltfImporter::doCamera(UnsignedInt id) {
+    const tinygltf::Camera& camera = _d->model.cameras[id];
+
+    Float far = Constants::nan();
+    Float near = Constants::nan();
+    Rad fov = Rad{Constants::nan()};
+
+    if(camera.type == "perspective") {
+        far = camera.perspective.zfar;
+        near = camera.perspective.znear;
+        fov = Rad{camera.perspective.yfov};
+    } else if(camera.type == "orthographic") {
+        far = camera.orthographic.zfar;
+        near = camera.orthographic.znear;
+    }
+
+    return CameraData{fov, near, far, &camera};
+}
+
+UnsignedInt TinyGltfImporter::doCameraCount() const {
+    return _d->model.cameras.size();
+}
+
+Containers::Optional<LightData> TinyGltfImporter::doLight(UnsignedInt id) {
+    const tinygltf::Light& light = _d->model.lights[id];
+
+    Color3 lightColor{float(light.color[0]), float(light.color[1]), float(light.color[2])};
+    Float lightIntensity{1.0f}; /* not supported by tinygltf */
+
+    LightData::Type lightType;
+    std::string type = "spot";
+
+    if(light.type == "point") {
+        lightType = LightData::Type::Point;
+    } else if(light.type == "spot") {
+        lightType = LightData::Type::Spot;
+    } else if(light.type == "directional") {
+        lightType = LightData::Type::Infinite;
+    } else if(light.type == "ambient") {
+        Error() << "Trade::TinyGltfImporter::light(): unsupported value for light type:" << type;
+        return Containers::NullOpt;
+    } else {
+        Error() << "Trade::TinyGltfImporter::light(): invalid value for light type:" << type;
+        return Containers::NullOpt;
+    }
+
+    return LightData{lightType, lightColor, lightIntensity, &light};
+}
+
+UnsignedInt TinyGltfImporter::doLightCount() const {
+    return _d->model.lights.size();
+}
+
+UnsignedInt TinyGltfImporter::doSceneCount() const { return _d->model.scenes.size(); }
+
+Containers::Optional<SceneData> TinyGltfImporter::doScene(UnsignedInt id) {
+    std::vector<UnsignedInt> children;
+    const tinygltf::Scene& scene = _d->model.scenes[id];
+
+    for(const int node : scene.nodes) {
+        children.push_back(node);
+    }
+    return SceneData{{}, children};
+}
+
+UnsignedInt TinyGltfImporter::doObject3DCount() const {
+    return _d->model.nodes.size();
+}
+
+std::string TinyGltfImporter::doObject3DName(UnsignedInt id) {
+    return _d->model.nodes[id].name;
+}
+
+Int TinyGltfImporter::doObject3DForName(const std::string& name) {
+    const auto found = _d->nodesForName.find(name);
+    return found == _d->nodesForName.end() ? -1 : found->second;
+}
+
+std::unique_ptr<ObjectData3D> TinyGltfImporter::doObject3D(UnsignedInt id) {
+    const tinygltf::Node& node = _d->model.nodes[id];
+
+    std::vector<UnsignedInt> children(node.children.begin(), node.children.end());
+
+    Matrix4 transformation;
+
+    if(node.rotation.size() == 4) {
```

Comment:
What if `node.rotation.size() == 1`, for example? Can that happen? Or is it always either zero or the proper size?
I would put some sanity assertions here, for example: 

```cpp
CORRADE_INTERNAL_ASSERT(node.rotation.size() == 0 || node.rotation.size() == 4);
CORRADE_INTERNAL_ASSERT(node.translation.size() == 0 || node.translation.size() == 3);
CORRADE_INTERNAL_ASSERT(node.scale.size() == 0 || node.scale.size() == 3);
```